### PR TITLE
newtype for athena Address

### PR DESCRIPTION
**Pull request by poszu**
Change `athena_interface::Address` to be a newtype wrapping a `[u8;24]` instead of a type alias. 

Benefits:
- stronger type safety,
- allows for implementing foreign traits (like `std::fmt::Display`, `From`)
- removed some unnecessary wrapper code (`AddressWrapper`)

### CODE REVIEW

File: interface/src/lib.rs [hunk 1]
```diff
@@ -337,11 +331,11 @@ pub fn calculate_address(
 ) -> Address {
   // calculate address by hashing the template, blob, principal, and nonce
   let mut hasher = Hasher::new();
-  hasher.update(template);
+  hasher.update(&template.0);
```
Comment (lrettig) [inline interface/src/lib.rs:334]: wouldn't an Into method for a byte slice make this more readable?

File: ffi/athcon/bindings/rust/athcon-client/tests/integration_tests.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@ use athcon_client::{
 };
 use athcon_vm::{MessageKind, Revision, StatusCode, StorageStatus};
 use athena_interface::payload::ExecutionPayload;
-use athena_interface::ADDRESS_ALICE;
+const ADDRESS_ALICE: Address = [0xAAu8; 24];
```
Comment (lrettig) [inline ffi/athcon/bindings/rust/athcon-client/tests/integration_tests.rs:11]: Why redefine this here?

File: vm/entrypoint/src/helpers/mod.rs [hunk 3]
```diff
@@ -1,20 +1,8 @@
-use athena_interface::{Address, Balance, Bytes32, Bytes32Wrapper};
+use athena_interface::Bytes32;
 use bytemuck::cast;
 
 // Simple helper functions for smart contracts
 
-pub fn address_to_bytes32(address: Address) -> Bytes32 {
```
Comment (lrettig) [inline vm/entrypoint/src/helpers/mod.rs:6]: I see, this is nice

**Review by jellonek — APPROVED** [review]

File: ffi/athcon/bindings/rust/athcon-client/tests/integration_tests.rs (hunk 2 above)
Comment (poszu) [reply ffi/athcon/bindings/rust/athcon-client/tests/integration_tests.rs:11]: I'm slowly removing all uses of the dummy addresses defined in the `athena-interface` with the end goal of removing them entirely. Every test should define the addresses it wants to work with itself.

File: interface/src/lib.rs (hunk 1 above)
Comment (poszu) [reply interface/src/lib.rs:334]: `Into` consumes the value.  I will add implementations of `AsRef` instead.

**Pull request merged**